subsys/net/lib/lwm2m/lwm2m_obj_swmgmt.c

```diff
@@ -731,19 +731,19 @@ static struct lwm2m_engine_obj_inst *swmgmt_create(uint16_t obj_inst_id)
 			  res_inst_idx, &instance->package_name, PACKAGE_NAME_LEN, 0);
 
 	INIT_OBJ_RES_LEN(SWMGMT_PACKAGE_VERSION_ID, res[index], res_idx, res_inst[index],
-			 res_inst_idx, 1, true, false, &instance->package_version,
+			 res_inst_idx, 1, false, true, &instance->package_version,
 			 PACKAGE_VERSION_LEN, 0, state_read_pkg_version, NULL, NULL, NULL, NULL);
 
 	INIT_OBJ_RES_OPT(SWMGMT_PACKAGE_ID, res[index], res_idx, res_inst[index], res_inst_idx, 1,
-			 true, false, NULL, NULL, package_write_cb, NULL, NULL);
+			 false, false, NULL, NULL, package_write_cb, NULL, NULL);
 
 #ifdef CONFIG_LWM2M_FIRMWARE_UPDATE_PULL_SUPPORT
 	INIT_OBJ_RES(SWMGMT_PACKAGE_URI_ID, res[index], res_idx, res_inst[index], res_inst_idx, 1,
-		     true, true, instance->package_uri, PACKAGE_URI_LEN, NULL, NULL, NULL,
+		     false, false, instance->package_uri, PACKAGE_URI_LEN, NULL, NULL, NULL,
 		     package_uri_write_cb, NULL);
 #else
 	INIT_OBJ_RES_OPT(SWMGMT_PACKAGE_URI_ID, res[index], res_idx, res_inst[index], res_inst_idx,
-			 1, true, false, NULL, NULL, package_uri_write_cb, NULL, NULL);
+			 1, false, false, NULL, NULL, package_uri_write_cb, NULL, NULL);
 #endif
```

Comment:
This makes the URI optional. That complies with the spec but is a bit of a harsh change. The user needs to remember to call 

```c
	lwm2m_engine_create_res_inst("/N/M/3/0");
```

Reply:
I'm not sure if we should do this. Yes, the spec makes the URI resource optional, but if you support PULL mode firmware download, then it makes little sense not to have it.

Reply:
@rlubos I guess the question is: is complying with the spec more important than easy of use? Don't have a strong opinion here. Feel free to tell me what it should be ;)

Reply:
I really don't have a feeling we're doing something against the spec here by instantiating optional resource **in certain conditions**. We only do it if PULL SW update mode is enabled, and I say it again - it makes little sense to have it w/o URI resource being available.

Reply:
@rlubos ok, fixed